### PR TITLE
Message id version 30

### PR DESCRIPTION
**Pull request by mthambipillai**

### CODE REVIEW

File: src/main/java/com/streamr/broker/kafka/KafkaRecordTransformer.java [hunk 1]
```diff
@@ -1,20 +1,28 @@
...
 import java.util.HashMap;
 import java.util.Map;
 
 class KafkaRecordTransformer {
 	private final Map<String, Long> offsetByStream = new HashMap<>();
 
-	StreamrBinaryMessageWithKafkaMetadata transform(ConsumerRecord<String, byte[]> record) {
-		StreamrBinaryMessage streamrBinaryMessage = StreamrBinaryMessage.from(ByteBuffer.wrap(record.value()));
+	StreamMessage transform(ConsumerRecord<String, byte[]> record) throws IOException {
+		String json = new String(record.value(), StandardCharsets.UTF_8);
```
Comment (hpihkala) [inline src/main/java/com/streamr/broker/kafka/KafkaRecordTransformer.java]: Danger of getting an exception here (?), if the content is binary which doesn't happen to be representable in UTF-8. (Sure, all UTF-8 strings have a binary representation, but not all binary might be valid UTF-8 strings!)

Would be safer to inspect the raw byte(s) of the `byte[]` to establish whether it's a `StreamrBinaryMessage` or a JSON message, and only parse to UTF-8 String if the latter.

File: src/test/groovy/com/streamr/broker/stats/LoggedStatsSpec.groovy [hunk 2]
```diff
@@ -60,9 +60,9 @@ class LoggedStatsSpec extends Specification {
 		recordedLogs == [
 			"Statistics logger started. Logging interval is 1 sec(s).",
 			"\n\tLast timestamp ${new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date(0))}\n" +
-				"\tBackpressure 0.17 kB / 4 events\n" +
-				"\tRead throughput 0.255 kB/s or 6 event/s\n" +
-				"\tWrite throughput 0.085 kB/s or 2 event/s\n" +
+				"\tBackpressure 0.966 kB / 4 events\n" +
```
Comment (hpihkala) [inline src/test/groovy/com/streamr/broker/stats/LoggedStatsSpec.groovy]: Did the `ExampleData` message sizes grow so considerably due to signatures added, or due to the binary -> json change? If the former, the growth seems expected, but if the latter, we might have a problem...

Comment (mthambipillai) [reply src/test/groovy/com/streamr/broker/stats/LoggedStatsSpec.groovy]: I think it's due to the additional fields. The changes in the messages in ExampleData is the following:
+ sequenceNumber (int here but will be long)
+ publisherId (String of 42 chars)
+ previousTimestamp (long)
+ previousSequenceNumber (int here but will be long)
+ signature (String of 132 chars)
- kafkaOffset (int)
- previousKafkaOffset (int)

File: src/main/java/com/streamr/broker/cassandra/CassandraStatementBuilder.java [hunk 3]
```diff
@@ -23,7 +23,7 @@ BoundStatement eventInsert(StreamMessage msg) {
 			msg.getStreamId(),
 			msg.getStreamPartition(),
 			msg.getTimestampAsDate(),
-			msg.getSequenceNumber(),
+			(int) msg.getSequenceNumber(),
```
Comment (hpihkala) [inline src/main/java/com/streamr/broker/cassandra/CassandraStatementBuilder.java]: Why cast to `int`?

Comment (hpihkala) [reply src/main/java/com/streamr/broker/cassandra/CassandraStatementBuilder.java]: `streamr-docker-dev` updated to have a longer field for sequence number in Cassandra

**Pull request merged**